### PR TITLE
Feature/6.4/tdi 38262 split salesforce

### PR DESCRIPTION
**Pull request by wwang-talend**
**Please check if the PR fulfills these requirements**

- [x] The commit message follows Talend standard
- [x] Tests for the changes have been added (for bug fixes / features)

**What is the current behavior?** (You can also link to an open issue here)
https://jira.talendforge.org/browse/TDI-38262
some part for https://jira.talendforge.org/browse/TDI-38201

**Does this PR introduce a breaking change?**

- [ ] Yes
- [x] No

### CODE REVIEW

File: components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcegetupdated/TSalesforceGetUpdatedDefinition.java [hunk 1]
```diff
@@ -45,7 +43,7 @@ public String getPartitioning() {
     public RuntimeInfo getRuntimeInfo(ExecutionEngine engine, ComponentProperties properties, ConnectorTopology componentType) {
         assertEngineCompatibility(engine);
         if (componentType == ConnectorTopology.OUTGOING) {
-            return getCommonRuntimeInfo(this.getClass().getClassLoader(), SalesforceSource.class);
+            return getCommonRuntimeInfo("org.talend.components.salesforce.runtime.SalesforceSource");
```
Comment (gonchar-ivan) [inline components/components-salesforce/components-salesforce-definition/src/main/java/org/talend/components/salesforce/tsalesforcegetupdated/TSalesforceGetUpdatedDefinition.java]: Could be a constant in parent SalesforceDefinition

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/206/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/206/)

**Comment by gonchar-ivan** [comment]
Could be merged. No Major/Critical issues.
There are some minor. Created Jira issues for them:
https://jira.talendforge.org/browse/TDI-38363
https://jira.talendforge.org/browse/TDI-38364
https://jira.talendforge.org/browse/TDI-38365

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/207/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/207/)

**Comment by wwang-talend** [comment]
the last pr to master as : https://ci-common.datapwn.com/job/components_pull-request_master/207/artifact/information_pull_request.txt/*view*/
4b6945e231e0fcb3eb5655cfa6c961433647a6ef is the previous commit's id, so it doesn't contains my test.

**Comment by wwang-talend** [comment]
have stopped the branch test, as it have no meaning, will trigger the pr to master test

**Comment by wwang-talend** [comment]
the test hold on as : https://ci-common.datapwn.com/job/components_pull-request_master/233/consoleFull

it already retry connect to : INFO: Retrying request to {s}->https://talend-update.talend.com:443

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/233/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/233/)

**Comment by build-talend-common** [comment]
Pull request test result: [failure](https://ci-common.datapwn.com/job/components_pull-request_master/234/) [:x:](https://ci-common.datapwn.com/job/components_pull-request_master/234/)

**Comment by wwang-talend** [comment]
all salesforce tests have been passed, only one BigQueryDatasetTestIT.testBasic fails which come from another project.

**Pull request merged**